src/absil/ilwrite.fs

```diff
@@ -2349,17 +2319,7 @@ let GenILMethodBody mname cenv env (il: ILMethodBody) =
 // -------------------------------------------------------------------- 
 
 let rec GetFieldDefAsFieldDefRow cenv env (fd: ILFieldDef) = 
-    let flags = 
-        GetMemberAccessFlags fd.Access |||
-        (if fd.IsStatic then 0x0010 else 0x0) |||
-        (if fd.IsInitOnly then 0x0020 else 0x0) |||
-        (if fd.IsLiteral then 0x0040 else 0x0) |||
-        (if fd.NotSerialized then 0x0080 else 0x0) |||
-        (if fd.IsSpecialName then 0x0200 else 0x0) |||
-        (if fd.IsSpecialName then 0x0400 else 0x0) ||| // REVIEW: RTSpecialName 
-        (if (fd.LiteralValue <> None) then 0x8000 else 0x0) |||
-        (if (fd.Marshal <> None) then 0x1000 else 0x0) |||
-        (if (fd.Data <> None) then 0x0100 else 0x0)
+    let flags = int fd.Attributes
```

Comment:
Likewise here ``ILFieldDef`` has duplicated information in both ``Attributes`` and ``Data``, ``LiteralValue`` and ``Offset``.  You either have to remove the duplication of information or be really sure that the attributes are updated when updating this information.  Adding a ``ilFielDef.WithData(d)`` update method is probably a good idea to ensure the consistency invariants are more obviously maintained?